ambry-store/src/main/java/com/github/ambry/store/StorageManager.java

```diff
@@ -211,6 +216,11 @@ public void start() throws InterruptedException, StoreException {
         });
       }
       diskToDiskManager.values().forEach(diskManager -> unexpectedDirs.addAll(diskManager.getUnexpectedDirs()));
+
+      // Add the background task to update the disk capacity
+      DiskFailureHandler diskFailureHandler = new DiskFailureHandler();
+      scheduler.scheduleAtFixedRate(diskFailureHandler, storeConfig.storeDiskFailureHandlerTaskIntervalInSeconds,
```

Comment:
Shall we only schedule the task if   if clusterMap.isDataNodeInFullAutoMode(currentNode) is true? Otherwise we will keep schedule one task but do nothing.

Reply:
FullAuto can be turned on in runtime. We want to make sure that if full auto is turned off, we don't run the handler.